src/hidpp20.c

```diff
@@ -2884,6 +2894,8 @@ hidpp20_device_new(const struct hidpp_device *base, unsigned int idx)
 	dev->proto_major = 1;
 	dev->proto_minor = 0;
 
+	hidpp_get_supported_report_types(&(dev->base), idx);
```

Comment:
we should ensure we have at least the long or short reports available here.

Reply:
I am not sure about that. There are a lot of features that don't require short reports. It would be sane for a device to not have short reports. Do you still want the check here, as opposed to being in `hidpp20_request_command_allow_error`?

Reply:
No, I mean, a device not supporting both short and long reports is not suppoorted, correct? So if `->report_suported == 0`, then we should bail out.

Reply:
Nevermind, I misread your first reply :disappointed:.